great_expectations/data_context/data_context.py

```diff
@@ -3206,7 +3208,9 @@ def test_yaml_config(
         class_name: Optional[str] = None,
         runtime_environment: Optional[dict] = None,
         pretty_print: bool = True,
-        return_mode: str = "instantiated_class",
+        return_mode: Union[
+            Literal["instantiated_class"], Literal["report_object"]
+        ] = "instantiated_class",
```

Comment:
A little ugly but when we have a limited number of options, let's limit the typing to just those options. This is probably a good candidate for an enum but opted to just use `Literal` to keep things simple for now.

Reply:
oo fancy ❤️

Reply:
this is from the extension library we added to our requirements a long time ago!